### DISCUSSION TITLE
How can I subscribe to topic?

### DISCUSSION THREAD

**Discussion by FamilyMan** [nuxt-community/firebase-module#565] [Q&A] [answered]
Apparently firebase can use something like:

    FirebaseMessaging.getInstance().subscribeToTopic("all")
                .addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        //Successfully Subscribed to Notification Service
                    }else{
                    //Subscription  Notification Service failed
                    }
                }

How can I subscribe to a topic using this module?

**Comment by FamilyMan** [comment, marked as answer]
There is no client-side API in the JavaScript SDK to subscribe to a topic. You'll instead need to use a (server-side) Admin SDK to accomplish this.

For example, the subscribeToTopic method you refer to exists in the Node.js SDK and in the Java SDK, but not in the JavaScript/Web or Android SDKs.

I recommend studying the documentation on subscribing a web client to a topic a bit further: https://firebase.google.com/docs/cloud-messaging/js/topic-messaging#subscribe_the_client_app_to_a_topic

**Reply by JTibblesSoftcon** [reply]
Thanks for this information. I've been scratching my head for days on this. 
Also, do we know why susbscribeToTopics isn't available on the client-side API? It makes sense to have included it as they do with their other available apis like getToken
